tests/e2e.rs

```diff
@@ -41,47 +57,54 @@ async fn w3c_vc_did_client_direct_post() {
         .with_presentation_definition(presentation_definition.clone())
         .with_request_parameter(ResponseMode::DirectPost)
         .with_request_parameter(ResponseType::VpToken)
-        .with_request_parameter(Nonce("random123".to_owned()))
+        .with_request_parameter(Nonce::random())
         .with_request_parameter(ClientMetadata(client_metadata))
         .build(wallet.metadata().clone())
         .await
         .unwrap();
 
+    println!("Request: {:?}", request);
+
     let request = wallet.validate_request(request).await.unwrap();
 
+    let parsed_presentation_definition = request
+        .resolve_presentation_definition(wallet.http_client())
+        .await
+        .unwrap();
+
     assert_eq!(
         &presentation_definition,
-        request
-            .resolve_presentation_definition(wallet.http_client())
-            .await
-            .unwrap()
-            .parsed()
+        parsed_presentation_definition.parsed()
     );
 
     assert_eq!(&ResponseType::VpToken, request.response_type());
 
     assert_eq!(&ResponseMode::DirectPost, request.response_mode());
 
-    // TODO: Response with a VP.
-    let presentation_submission: PresentationSubmission = serde_json::from_value(json!(
-        {
-            "id": "39881a17-e454-4d98-87ba-e3073d1014d6",
-            "definition_id": "0b4dd017-efa6-4a05-a269-9790fa3c22c2",
-            "descriptor_map": [
-                {
-                    "id": "064255a8-a0fa-4108-9ded-429f83003350",
-                    "path": "$",
-                    "format": "jwt_vp"
-                }
-            ]
-        }
-
-    ))
-    .unwrap();
+    let descriptor_map = parsed_presentation_definition
+        .parsed()
+        .input_descriptors()
+        .iter()
+        .map(|descriptor| {
+            // NOTE: the input descriptor constraint field path is relative to the path
+            // of the descriptor map matching the input descriptor id.
+            DescriptorMap::new(
+                descriptor.id().clone(),
+                ClaimFormatDesignation::JwtVc,
```

Comment:
Add the following code comment:

```
// NOTE: Since the input descriptor may support several different claim format types. This value should not be 
// hardcoded in production code, but should be selected from available formats in the presentation definition 
// input descriptor.
```